streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractDualSchemaRocksDBSegmentedBytesStore.java

```diff
@@ -243,15 +246,8 @@ public String name() {
     public void init(final StateStoreContext stateStoreContext, final StateStore root) {
         this.internalProcessorContext = asInternalProcessorContext(stateStoreContext);
 
-        final StreamsMetricsImpl metrics = ProcessorContextUtils.metricsImpl(stateStoreContext);
-        final String threadId = Thread.currentThread().getName();
-        final String taskName = stateStoreContext.taskId().toString();
-
-        expiredRecordSensor = TaskMetrics.droppedRecordsSensor(
-            threadId,
-            taskName,
-            metrics
-        );
+        streamsMetrics = ProcessorContextUtils.metricsImpl(stateStoreContext);
```

Comment:
I don't think we need to add these two member variables? We keep a reference of `stateStoreContext` as `internalProcessorContext`, and should be able to use `internalProcessorContext` inside `initMetricsIfNeeded` to get `taskId` and the `streamsMetrics` object?